src/emmodel/model.py

```diff
@@ -126,4 +126,75 @@ def plot_time_trend(ax: plt.Axes, df: pd.DataFrame,
     ax.set_xlabel("time")
     for time in year_heads:
         ax.axvline(time, linestyle="--", color="gray")
-    return ax
\ No newline at end of file
+    return ax
+
+
+class LinearExcessMortalityModel:
+    """LinearExcessMortalityModel"""
+    def __init__(self, df: pd.DataFrame, col_obs: str, 
+                 stage1_col_covs: List[str], stage2_col_covs: List[str],
+                 col_covs_priors: List[str], model_type: str = 'two-stage'):
+        super(LinearExcessMortalityModel, self).__init__()
+        self.df = df
+        self.col_obs = col_obs
+        self.stage1_col_covs = stage1_col_covs
+        self.stage2_col_covs = stage2_col_covs
+        self.model_type = model_type
+        self.col_covs_priors = col_covs_priors
+
+    def run(self, prior_min_var: float = 0.1):
+        if self.model_type == 'two-stage':
+            self.run_stage_1()
+            self.priors = self.result_to_priors(min_var=prior_min_var)
+            self.run_stage_2()
+
+    def run_stage_1(self):
+        self.data = regmod.data.Data(col_obs=self.col_obs, col_covs=self.stage1_col_covs)
+        self.variables = [
+            regmod.variable.Variable(self.stage1_col_covs[ix]) 
+                for ix in range(len(self.stage1_col_covs))
+            ]
+        self.data.attach_df(self.df)
+        self.model = regmod.model.LinearModel(self.data, self.variables)
+        self.result = regmod.optimizer.scipy_optimize(self.model)
+        self.data.detach_df()
+
+    def run_stage_2(self):
+        self.data = regmod.data.Data(col_obs=self.col_obs, col_covs=self.stage2_col_covs)
+        dropna_col_covs = self.stage2_col_covs.copy()
+        if 'intercept' in dropna_col_covs:
+            dropna_col_covs.remove('intercept') 
+        self.data.attach_df(self.df.dropna(subset=dropna_col_covs))
+
+        self.variables = []
+        for cov in self.col_covs_priors:
+            self.variables.append(regmod.variable.Variable(cov, priors=[self.priors[cov]]))
+
+        for cov in self.stage2_col_covs:
+            if cov not in self.col_covs_priors:
+                self.variables.append(regmod.variable.Variable(cov))
+
+        self.model = regmod.model.LinearModel(self.data, self.variables)
+        self.result = regmod.optimizer.scipy_optimize(self.model)
+        self.data.detach_df()
+
+    def result_to_priors(self, min_var: float = 0.1):
```

Comment:
for priors here we should remove the prior on `intercept` since we want the `intercept` to still fit freely in the second stage.

Reply:
sure

Reply:
I also didn't use priors for intercept, whether to use priors is controlled by col_covs_priors